tools/populate-owners/main.go

```diff
@@ -0,0 +1,323 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	doNotEdit            = "# DO NOT EDIT; this file is auto-generated using tools/populate-owners.\n"
+	ownersComment        = "# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md\n"
+	ownersAliasesComment = "# See the OWNERS_ALIASES docs: https://git.k8s.io/community/contributors/guide/owners.md#owners_aliases\n"
+)
+
+// owners is copied from k8s.io/test-infra/prow/repoowners's Config
+type owners struct {
+	Approvers         []string `json:"approvers,omitempty" yaml:"approvers,omitempty"`
```

Comment:
This approach has more structure than Python, but unfortunately the `OWNERS` content doesn't seem to be particularly structured.  I've based this structure on [`Config`][1], but there's also `options` and `filters` property declared [in `FullConfig`][2].  And in this repo, there's a `filters` consumer [here][3].  Some upstream `OWNERS` also [use `features` and `aliases`][4], and I haven't tracked down the origin of those.  Are we worried about preserving those sorts of things during this copy?  Should I declare new properties as we hit them, or should I just make `owners` an `interface{}` to ensure we preserve all upstream properties?

[1]: https://github.com/kubernetes/test-infra/blob/204ae03140562a49b33412b1057459adf55b1bc6/prow/repoowners/repoowners.go#L51-L57
[2]: https://github.com/kubernetes/test-infra/blob/204ae03140562a49b33412b1057459adf55b1bc6/prow/repoowners/repoowners.go#L70-L74
[3]: https://github.com/openshift/release/blob/94dbba47fdde35b6d38f287c19130c25de22cedb/ci-operator/templates/OWNERS#L1
[4]: https://github.com/openshift/coredns/blob/5e0206e86b16837461244d0f0c5d7c19de8e1f4e/OWNERS#L24-L32